File: api/tasks.py
import sys
sys.path.append("..")
import eel
from models import Task
from api.attachments import delete_all_attachments, delete_attachment

# tasks api

@eel.expose
def create_task(name, complete, folder_id):
    task = Task.create(name=name, complete=complete, folder=folder_id)
    return task.id


@eel.expose
def delete_task(id):
    task = Task.get(Task.id == id)
    delete_all_attachments(task.id, 'task')
    return task.delete_instance()
    
@eel.expose
def get_task(id):
    task = Task.get(Task.id == id)
    return {'id': task.id, 'name': task.name, 'complete': task.complete}

@eel.expose
def get_tasks(folder_id):
    tasks = Task.select().where(Task.folder_id == folder_id)
    return [{'id': task.id, 'name': task.name, 'time': task.time, 'complete': task.complete} for task in tasks]


@eel.expose
def update_task(id, complete, folder_id):
    task = Task.get(Task.id == id)
    task.complete = complete
    if folder_id != '':
        task.folder = folder_id
    task.save()

File: web/js/components/tasks.js
async function createTask() {
  const title = document.getElementById('taskName').value;
  const folder = getCookie('folder');
  const id = await eel.create_task(title, false, folder)();
  await getTasks(folder);
}

async function getTasks(folder) {
  const tasks = await eel.get_tasks(folder)();
  let formAction = `sendTemplate({'setName': 'taskName', 'place': 'to do', 'action': 'createTask()', 'button': 'new task'}, 'smallForm', 'home', 'formHolder')`;
  let html = await loadTemplate({ 'formAction': formAction, 'title': 'tasks', 'icon': 'bi bi-clipboard2-check-fill'}, 'sectionHeader', 'folders');
  html += await eel.render_template_list(tasks, 'tasks', 'items')();
  document.getElementById('tasks').innerHTML = html;
  if (getCookie('hidetasks') != 'true') {
    for (let i = 0; i < tasks.length; i++) {
      let task = tasks[i]
      let check = task.complete;
      document.getElementById('taskButton' + task.id).checked = check;
    }
  }

  
}

async function checkTask(id) {
  let complete = document.getElementById('taskButton' + id).checked;
  await eel.update_task(id, complete, '')();
  await eel.get_tasks(getCookie('folder'))();
}

async function deleteTask(id) {
  await eel.delete_task(id)();
  await getTasks(getCookie('folder'));
}

async function reloadTask(id, name, time, template) {
  complete = document.getElementById('taskButton' + id).checked;
  sendTemplate({ 'name': name, 'time': time, 'complete': complete, 'id': id }, template, 'tasks', 'task' + id)
}